homeassistant/components/esphome/ffmpeg_proxy.py

```diff
@@ -201,6 +201,9 @@ async def transcode(
         write_task = self.hass.async_create_background_task(
             self._write_ffmpeg_data(request, writer, proc), "ESPHome media proxy"
         )
+        self.hass.async_create_background_task(
+            self._dump_ffmpeg_stdout(proc), "ESPHome media proxy dump stderr"
+        )
```

Comment:
If there is an exception in `_write_ffmpeg_data` both tasks will be competing to consume `proc.stderr`

Reply:
Maybe start the task in `_write_ffmpeg_data`, hold a reference to it, and cancel it `except` and `finally` block.

unrelated, but that bare `except:` will swallow `KeyboardInterrupt` and `SystemExit`

Reply:
the task which reads the ffmpeg stderr is now started and canceled within `_write_ffmpeg_data`. to avoid competing to consume `stderr`, `_write_ffmpeg_data` will not consume it anymore.

I've tested with `kill -15 <ffmpeg-pid>` and `kill -9 <ffmpeg-pid>` to trigger the vare `except` in `_write_ffmpeg_data`, but it never got triggered